zipline/assets/assets.py

```
@@ -506,14 +627,17 @@ def map_identifier_index_to_sids(self, index, as_of_date):
         # Return a list of the sids of the found assets
         return [asset.sid for asset in matches]
 
-    def insert_metadata(self, identifier, **kwargs):
+    def _insert_metadata(self, identifier, **kwargs):
         """
         Inserts the given metadata kwargs to the entry for the given
         identifier. Matching fields in the existing entry will be overwritten.
         :param identifier: The identifier for which to insert metadata
         :param kwargs: The keyed metadata to insert
         """
-        entry = self.metadata_cache.get(identifier, {})
+        if identifier in self.metadata_cache:
+            # Multiple pass insertion no longer supported.
+            return
```

Should we throw an error/warning on this to avoid confusion?